### DISCUSSION TITLE
I get type error while trying to call sqsClient with a configuration object.

### DISCUSSION THREAD

**Discussion by palashCItobuz** [aws/aws-sdk-js-v3#5706] [Q&A] [answered]
I am trying to initialize the sqsClient
```
const sqs = new SQSClient({
  apiVersion: process.env.AWS_SQS_APIVERSION,
  credentials: {
    accessKeyId: process.env.AWS_ACCESSKEY,
    secretAccessKey: process.env.AWS_SECRETKEY,
  },
  region: process.env.AWS_S3_REGION
});
```
However, I get this type related problem while trying to initialize. 
<img width="918" alt="image" src="https://github.com/aws/aws-sdk-js-v3/assets/7223468/f1008230-ded3-4993-8181-6fc5d6632512">
It shows the type accepts an array of configuration objects.

```
export class SQSClient extends __Client {
    constructor(...[configuration]) {
```
When I wrap the configuration object in an array, the error disappears.

**Comment by RanVaknin** [comment, marked as answer]
Hi @palashCItobuz ,

The typescript compiler complains about those values being potentially undefined. 

You can assert that they are not null by using an if block:

```js
if (process.env.AWS_ACCESSKEY && process.env.AWS_SECRETKEY){
  const sqs = new SQSClient({
    apiVersion: process.env.AWS_SQS_APIVERSION,
    credentials: {
      accessKeyId: process.env.AWS_ACCESSKEY,
      secretAccessKey: process.env.AWS_SECRETKEY,
    },
    region: process.env.AWS_S3_REGION
  });
} else {
  throw new Error("SDK cannot be initialized with undefined credentials")
}
```

Or simplified by using the pound (`!`) operator to assert non-null:

```js
const sqs = new SQSClient({
  apiVersion: process.env.AWS_SQS_APIVERSION,
  credentials: {
    accessKeyId: process.env.AWS_ACCESSKEY!,
    secretAccessKey: process.env.AWS_SECRETKEY!,
  },
  region: process.env.AWS_S3_REGION!
});
```

Let me know if this helps.
Thanks,
Ran~

**Reply by palashCItobuz** [reply]
@RanVaknin It works. But how does it work when I pass the object wrapped up in an array without any conditional check? 

```
new S3Client([{
  region: process.env.AWS_ACCESSKEY,
  credentials: {
    accessKeyId: process.env.AWS_ACCESSKEY,
    secretAccessKey: process.env.AWS_SECRETKEY,
  }
}]);
```
<img width="587" alt="image" src="https://github.com/aws/aws-sdk-js-v3/assets/7223468/d967f50c-0487-4d2c-abc1-f337a2ab5ec1">
